libs/core/src/lib/flexible-column-layout/flexible-column-layout.component.ts

```diff
@@ -0,0 +1,421 @@
+
+  /**
+   * the layout of the component
+   * Options include: 'OneColumnStartFullScreen' | 'OneColumnMidFullScreen' |
+   * 'OneColumnEndFullScreen' | 'TwoColumnsStartExpanded' | 'TwoColumnsMidExpanded' |
+   * 'TwoColumnsEndExpanded' | 'ThreeColumnsMidExpanded' | 'ThreeColumnsEndExpanded' |
+   * 'ThreeColumnsStartMinimized' | 'ThreeColumnsEndMinimized'
+   */
+  @Input()
+  layout: Layout = ONE_COLUMN_START_FULL_SCREEN;
+
+  /**
+   * the event emitted on layout value cchange
+   */
+  @Output()
+  layoutChange: EventEmitter<Layout> = new EventEmitter<Layout>();
+
+  /**
+   * user defined break point for MD screens
+   * the default (Fiori 3) value is 599
+   */
+  @Input()
+  mdBreakPoint = 599;
+
+  /**
+   * user defined break point for LG screens
+   * the default (Fiori 3) value is 1023
+   */
+  @Input()
+  lgBreakPoint = 1023;
+
+  /**
+   * user defined break point for XL screens
+   * the default (Fiori 3) value is 1439
+   */
+  @Input()
+  xlBreakPoint = 1439;
+
+  /**
+   * user defined onResize function
+   */
+  @Input()
+  customOnResizeFunction: () => void;
+
+  /**
+   * left column separator value (between start and middle columns)
+   * that specifies the direction of the arrow and
+   * if the separator is visible
+   * Options include: 'left', 'right' and null
+   */
+  _leftColumnSeparator: ColumnSeparatorValue = null;
+
+  /**
+   * right column separator value (between middle and end columns)
+   * that speccifies the direction of the arrow and
+   * if the separator is visible
+   * Options include: 'left', 'right' and null
+   */
+  _rightColumnSeparator: ColumnSeparatorValue = null;
+
+  /** 
+   * @hidden 
+   * allows to keep track of the previos layout 
+   * so we can go back to it on window resize
+   */
+  private _previousLayout: Layout = this.layout;
+
+  /** @hidden */
+  private _screenSize: ScreenSize = LG_SCREEN_SIZE;
+
+  /** @hidden */
+  private _subscriptions = new Subscription();
+
+  /** 
+   * @hidden 
+   * mappting of the layout name and the column layout in %
+   */
+  private _layouts = {
+    OneColumnStartFullScreen: { start: 100, mid: 0, end: 0 },
+    OneColumnMidFullScreen: { start: 0, mid: 100, end: 0 },
+    OneColumnEndFullScreen: { start: 0, mid: 0, end: 100 },
+    TwoColumnsStartExpanded: { start: 67, mid: 33, end: 0 },
+    TwoColumnsMidExpanded: { start: 33, mid: 67, end: 0 },
+    TwoColumnsEndExpanded: { start: 0, mid: 33, end: 67 },
+    ThreeColumnsMidExpanded: { start: 25, mid: 50, end: 25 },
+    ThreeColumnsEndExpanded: { start: 25, mid: 25, end: 50 },
+    ThreeColumnsStartMinimized: { start: 0, mid: 67, end: 33 },
+    ThreeColumnsEndMinimized: { start: 33, mid: 67, end: 0 }
+  };
+
+  /**
+   * the column layout representing the distribution of the width
+   * between the first (start), the middle and the last(end) column
+   */
+  _columnLayout: { start: number; mid: number; end: number; } = this._layouts[this.layout];
```

Comment:
I think for this object should be created some interface, eg.
`export interface FlexibleColumnSettings { start: number; mid: number; end: number; }`